### ISSUE TITLE
Add Revision support

### ISSUE THREAD

**Issue by jasonbahl** [wp-graphql/wp-graphql#326]
The core WP REST API supports revisions: https://developer.wordpress.org/rest-api/reference/post-revisions/

They require the user to be authenticated, so we should respect similar auth patterns the REST API has established, but revisions should be able to be accessed from the core WPGraphQL Schema. 

I think it should be a field of the PostObjectType like so, something to this tune:

```
{
    post( id:"..." ) {
       id
       title
       revisions( first: 5 ) {
          edges {
            node {
              id
              title
           }
         }
      }
   }
}
```

I think there should also be a root query to get a single revision, like so, if you know the ID of the revision: 
```
{
   revision( id: "..." ) {
      id
      title
  }
}
```

Or you could get a revision from other parameters, such as the revisedContent type and ID.
```
{
   revision( where: { revisedContent: { type: POST id: "..." } } ) {
       id
       title
   }
}
```

**Comment by justinwhall** [comment]
This would be awesome :) 

It would facilitate building a Gatsby client route to preview posts. I'm assuming something like the following would retrieve the most recent revision:

```
{
    post( id:"..." ) {
       id
       title
       revisions( first: 1 ) {
          edges {
            node {
              id
              title
           }
         }
      }
   }
}
```

**Comment by jasonbahl** [comment]
@justinwhall have you used Gatsby 2.0 with WPGraphQL?

I haven't had a chance to work with Gatsby 2.0, but it looks like you can use any GraphQL source now, eh?

**Comment by jasonbahl** [comment]
closed by #695
